File: products/signals/eval/self_driving/tasks/signup-email-normalize/repo/acme_accounts/analytics.py
"""Fire-and-forget PostHog capture over plain HTTP (no SDK dependency)."""

import os
import json
import urllib.request

_HOST = os.environ.get("POSTHOG_HOST", "https://us.i.posthog.com")


def capture(distinct_id: str, event: str, properties: dict | None = None) -> None:
    api_key = os.environ.get("POSTHOG_API_KEY", "")
    if not api_key:
        return
    payload = json.dumps(
        {"api_key": api_key, "event": event, "distinct_id": distinct_id, "properties": properties or {}}
    ).encode()
    request = urllib.request.Request(f"{_HOST}/i/v0/e/", data=payload, headers={"Content-Type": "application/json"})
    try:
        urllib.request.urlopen(request, timeout=2)
    except OSError:
        pass  # Analytics must never take the request path down

File: products/signals/eval/self_driving/tasks/upload-mime-sniff/repo/src/analytics.js
let client = null

export async function initAnalytics() {
  const apiKey = process.env.POSTHOG_API_KEY
  if (!apiKey) {
    return
  }
  const { PostHog } = await import('posthog-node')
  client = new PostHog(apiKey, { host: process.env.POSTHOG_HOST ?? 'https://us.i.posthog.com' })
}

export function capture(distinctId, event, properties = {}) {
  client?.capture({ distinctId, event, properties })
}

export async function shutdownAnalytics() {
  await client?.shutdown()
  client = null
}
